sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -609,6 +609,9 @@ case class InSet(child: Expression, hset: Set[Any]) extends UnaryExpression with
   require(hset != null, "hset could not be null")
 
   override def toString: String = {
+    if (!child.resolved) {
+      return s"$child INSET " + hset.toSeq.mkString("(", ", ", ")")
```

Comment:
I wonder what we will see if `hset` elements are arrays like `Array[Byte]` - just address of the array, for timestamp - some number of microseconds?

Reply:
I think so, without data type there is not much we can do.

Reply:
Just don't output the values if their data type is not available.

Reply:
hmm, isn't it better to see the raw values than nothing?

Reply:
How about we don't output the values for non resolved child? Cause in normal cases, the query finally gets resolved and it will output values with resolved datatypes. @cloud-fan @MaxGekk